### PR TITLE
fix robot not updating status after sample load/unload

### PR DESCRIPTION
**Pull request by JunAishima**
* check that sample is mounted/unmounted as we expect after a mount/
   unmount is performed
 * replaces incorrect use of value from RE() - that returns a tuple of runIDs of
   plans executed
 * refs #182

### CODE REVIEW

File: denso_robot.py [hunk 1]
```diff
@@ -65,3 +65,21 @@ def unmount(self, gov_robot, puck_pos: int, pin_pos: int, samp_id: str):
...
     def finish(self):
         ...
+
+    def check_sample_mounted(self, mount, puck_pos, pin_pos):  # is the correct sample present/absent as expected during a mount/unmount?
+        if mount:
+            check_occupied = 1
+        else:
+            check_occupied = 0
+        if int(self.robot.spindle_occupied_sts.get()) == check_occupied and \
+           int(self.robot.puck_num_sel.get()) == puck_pos - 1 and \
```
Comment (RobertSchaffer1) [inline denso_robot.py:75]: Is this the zero indexed vs. one indexed issue?

Comment (JunAishima) [reply denso_robot.py:75]: yes - could add comments in that direction for all of the comparisons

**Review by mskinner5278 — APPROVED** [review]

**Comment by RobertSchaffer1** [comment]
Comment above:

I don't understand the TODO here, isn't that what this line is doing?

**Comment by JunAishima** [comment]
clarifying, there is a lot of status info within the EMBL robot code, which currently is just assumed to be successful. RE() does not return the function call value, so the various status codes need to be re-determined outside of RE() or replaced with a similar check as DensoRobot.

**Comment by RobertSchaffer1** [comment]
So shouldn't it be "TODO: don't assume embl robot is successful" or something like that?

**Comment by JunAishima** [comment]
I would update it, but already merged - it's more like "TODO: implement status when installed on AMX/FMX"

**Comment by RobertSchaffer1** [comment]
Hm, alright, something to remember then.

**Pull request merged**